packages/@react-spectrum/sidenav/test/SideNav.test.js

```diff
@@ -90,4 +124,90 @@ describe('SideNav', function () {
     expect(sideNavListItemLinks.length).toBe(4);
     expect(sideNavListItemLinks[0].getAttribute('target')).toBe('_self');
   });
+
+  it.each`
+    Name           | Component    | ComponentSection       | ComponentItem | props
+    ${'SideNav'}   | ${SideNav}   | ${Section}             | ${Item}       | ${{UNSAFE_className: 'test-class'}}
+    ${'V2SideNav'} | ${V2SideNav} | ${V2SideNavItem} | ${V2SideNavItem}}   | ${{className: 'test-class'}}
+  `('$Name can pass className', async function ({Name, Component, ComponentSection, ComponentItem, props}) {
+    let {getByRole} = renderComponent(Name, Component, ComponentSection, ComponentItem, props);
+
+    if (Component === SideNav) {
+      // wait for v3 components
+      await waitForDomChange();
+    }
+
+    let sideNav = getByRole('navigation');
+    expect(sideNav).toHaveAttribute('class', expect.stringContaining('test-class'));
+  });
+
+  // omitting v2 component
+  it.each`
+    Name                           | Component    | ComponentSection | ComponentItem
+    ${'SideNav'}                   | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavStatic'}             | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavWithSections'}       | ${SideNav}   | ${Section}       | ${Item}
+    ${'SideNavStaticWithSections'} | ${SideNav}   | ${Section}       | ${Item}
+  `('$Name can select an item', async function ({Name, Component, ComponentSection, ComponentItem}) {
+    let spy = jest.fn();
+    let {getByText} = renderComponent(Name, Component, ComponentSection, ComponentItem, {onSelectionChange: spy});
+
+    await waitForDomChange();
+
+    let [foo, bar, bob, alice] = [
+      getByText('Foo'),
+      getByText('Bar'),
+      getByText('Bob'),
+      getByText('Alice')
+    ];
+
+    bar.focus();
```

Comment:
I tried using just triggerPress... but I think that our triggerPress function is not focusing elements, so focus() needs to be triggered also with triggerPress

https://testing-library.com/docs/guide-events suggest something like this, since non of mouseDown, mouseUp or click will trigger focus:

fireEvent.mouseDown(element)
element.focus() (if that element is focusable) -> triggerPress miss this line
fireEvent.mouseUp(element)
fireEvent.click(element)

is it OK too add new function in utils: 
```
export function triggerPressWithFocus(element) {
  fireEvent.mouseDown(element);
  element.focus()
  fireEvent.mouseUp(element);
  fireEvent.click(element);
}
```

btw, needed focus to test tabIndex on element